test/cmake/seqan3_test_component.cmake

```diff
@@ -0,0 +1,39 @@
+# -----------------------------------------------------------------------------------------------------
+# Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+# Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+# This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+# shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+# -----------------------------------------------------------------------------------------------------
+
+# Get a specific component of a test file which follows the seqan3 naming scheme.
+# e.g. target_source_file = "range/views/take.cpp"
+# component:
+#  * TARGET_NAME - the target name (e.g. "take")
+#  * TARGET_UNIQUE_NAME - the target name which includes the target_path (e.g. range-view-take)
```

Comment:
What is the purpose of the target_unique_name?

Reply:
Like it said: the target name which includes the target_path (e.g. range-view-take)

```cmake
macro (seqan3_test_component TEST_TARGET_NAME "range/views/take.cpp" TARGET_NAME)
macro (seqan3_test_component TEST_TARGET_UNIQUE_NAME "range/views/take.cpp" TARGET_UNIQUE_NAME)
mesage(STATUS "TEST_TARGET_NAME: ${TEST_TARGET_NAME}") // take
mesage(STATUS "TARGET_UNIQUE_NAME: ${TARGET_UNIQUE_NAME}") // range-views-take
```